editor/editor_node.cpp

```diff
@@ -6867,8 +6883,25 @@ EditorNode::EditorNode() {
 	EDITOR_DEF("interface/inspector/default_color_picker_shape", (int32_t)ColorPicker::SHAPE_OKHSL_CIRCLE);
 	EditorSettings::get_singleton()->add_property_hint(PropertyInfo(Variant::INT, "interface/inspector/default_color_picker_shape", PROPERTY_HINT_ENUM, "HSV Rectangle,HSV Rectangle Wheel,VHS Circle,OKHSL Circle", PROPERTY_USAGE_DEFAULT));
 
+	EDITOR_DEF("system/environment_variables", Dictionary());
+	EditorSettings::get_singleton()->add_property_hint(PropertyInfo(Variant::DICTIONARY, "system/environment_variables", PROPERTY_HINT_ARRAY_TYPE, vformat("%d:%d", Variant::STRING, Variant::STRING), PROPERTY_USAGE_DEFAULT));
```

Comment:
```suggestion
	EditorSettings::get_singleton()->add_property_hint(PropertyInfo(Variant::DICTIONARY, "system/environment_variables", PROPERTY_HINT_TYPE_STRING, vformat("%d:%d", Variant::STRING, Variant::STRING), PROPERTY_USAGE_DEFAULT));
```

This has never been implemented for dictionaries. I think the `hint_string` format should support the ability to specify `hint` and `hint_string` key and value types. And also nesting. Something like:

```gdscript
"{%d:|%d:}" % [key_type, val_type]
"{%d/%d:%s|%d/%d:%s}" % [key_type, key_hint, key_hint_string, val_type, val_hint, val_hint_string]

# Nesting:
"{%d:|{%d:|%d:}}" % params3
"{%d/%d:%s|{%d/%d:%s|%d/%d:%s}}" % params9
```

Reply:
If this is tedious for you, then I am interested in implementing it. :)

Reply:
> `"{%d/%d:%s|%d/%d:%s}"`

The type hint for the `Dictionary` might benefit for a more complex format, at least should support multiple types for both keys and values (or even be something more like an XML schema, but this might be an overkill).

In any case, I have removed this part for this PR for now.

Reply:
Also, https://github.com/godotengine/godot/pull/78656 implements the same thing.